CMakeLists.txt

```diff
@@ -1,31 +1,20 @@
-cmake_minimum_required(VERSION 2.8.3)
-project(serial)
+cmake_minimum_required(VERSION 3.5.0)
```

Comment:
Why such an old version? I'm not sure what non-EOL platforms these days ship anything older than 3.16.

Reply:
RHEL8 ships with 3.16, the oldest version of CMake in a distro currently supported by a tier 1 or 2 versions of ROS to give you more context. @ChrisThrasher, do you mind helping explain the benefits of using a newer version of CMake and provide some references in case @moriarty would like more information about this recommendation?

Reply:
The main benefit is that it means you opt-in to many new and improved default behaviors (what CMake calls "policies") that have been added between version 3.5 and 3.16. 

https://cmake.org/cmake/help/latest/manual/cmake-policies.7.html

On top of that, nobody is actually using 3.5 or testing with 3.5 so while we can claim this script doesn't use features added after 3.5 we can't be sure that unless we add a CI job that uses 3.5. CMake will let you use newer features even if your minimum version is very old.

Reply:
I took 3.5 because I got a CMake warning that said something and suggested using at least 3.5

Reply:
Also when I saw that warning, I just cherr-picked because it was on the other open PR which is what is being released into debian as `libcxx-serial-dev` : [c9da89d](https://github.com/wjwwood/serial/pull/285/commits/c9da89d34893fbe975153795c158611873ab14fd) but when I tried to use that package I had to add some define around the includes of serial (WIP testing here:  https://github.com/PickNikRobotics/ros2_robotiq_gripper/pull/22)

Reply:
https://github.com/wjwwood/serial/pull/285/commits/7b7f62b606db7d661f46a3fe251134c8cd393a11 I've set this to 3.16